tv.kodi.Kodi.json

```diff
@@ -0,0 +1,128 @@
+{
+    "app-id": "tv.kodi.Kodi",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.openjdk9"
+    ],
+    "finish-args": [
+        "--device=dri",
+        "--filesystem=home",
+        "--share=ipc",
+        "--share=network",
+        "--socket=pulseaudio",
+        "--socket=wayland",
+        "--socket=x11",
+        "--system-talk-name=org.bluez",
+        "--system-talk-name=org.freedesktop.Avahi"
+    ],
+    "command": "kodi",
+    "post-install": [
+        "sed -i 's|^KODI_DATA.*|KODI_DATA=${HOME}/.var/app/tv.kodi.Kodi/data|' /app/bin/kodi"
+    ],
+    "modules": [
+        "deps/avahi.json",
+        "deps/bluez.json",
+        "deps/fribidi.json",
+        "deps/fstrcmp.json",
+        "deps/ghostscript.json",
+        "deps/glu.json",
+        "deps/libass.json",
+        "deps/libcdio.json",
+        "deps/lzo2.json",
+        "deps/samba.json",
+        "deps/swig.json",
+        "deps/taglib.json",
+        "deps/tinyxml.json",
+        "deps/udev.json",
+        {
+            "name": "kodi",
+            "buildsystem": "cmake",
+            "config-opts": [
+                "-DVERBOSE=1",
+                "-DENABLE_INTERNAL_FFMPEG=ON",
+                "-DENABLE_INTERNAL_FLATBUFFERS=ON",
+                "-DENABLE_INTERNAL_FMT=ON",
+                "-DENABLE_INTERNAL_RapidJSON=ON",
+                "-DFLATBUFFERS_URL=build/download/flatbuffers-1.9.0.tar.gz",
+                "-DCROSSGUID_URL=build/download/crossguid-8f399e8bd4.tar.gz",
+                "-DFMT_URL=build/download/fmt-5.1.0.tar.gz",
+                "-DRapidJSON_URL=build/download/rapidjson-1.1.0.tar.gz",
+                "-DFFMPEG_URL=build/download/4.0.2-Leia-Alpha3.tar.gz",
+                "-DLIBDVDCSS_URL=build/download/1.4.1-Leia-Beta-3.tar.gz",
+                "-DLIBDVDREAD_URL=build/download/6.0.0-Leia-Alpha-3.tar.gz",
+                "-DLIBDVDNAV_URL=build/download/libdvdnav-6.0.0-Leia-Alpha-3.tar.gz",
+                "-DJava_JAVA_EXECUTABLE=/usr/lib/sdk/openjdk9/bin/java"
+            ],
```

Comment:
You can make FFmpeg, FMT, FlatBuffers and RapidJson as separate modules. Please do this, else the build process can get really slow (their are build with make without the parameter -j<number>). CMake will find them automatically.

FFmpeg: You can use the official package from FFmpeg. IMPORTANT: Activate `--enable-vaapi`, `--enable-vdpau` for GPU speedup and `--enable-postproc` (Kodi search for this library) and `--enable-gpl` (else postproc will be not compiled and get automatically deactivated!!!).
FMT: Current lib is at 5.2.1 https://github.com/fmtlib/fmt/releases
FlatBuffers: Current lib is at 1.10.0 https://github.com/google/flatbuffers/releases
RapidJSON: Use that from Kodi mirror. I had problems during building the original tag from Tencent.

Reply:
We recommend using our ffmpeg. Just try to stick to whatever we define

Reply:
I checked the source out. The huge difference is that the Kodi version has a .gitignore file, rename it's version to Kodi and has deleted the file VERSION. Here is a diff https://gist.github.com/Marko10-000/80dbd05e9e67507ffcf1e8d8d34392d6 . But @MartijnKaijser is right, please use the file from the mirror, the rest remains the same.

Reply:
General note: If you feel Kodi uses a dependency that's too old please send a PR to get it updated.

Reply:
@MartijnKaijser is absolutely right, please use the file from the mirror. Here is a diff https://gist.github.com/Marko10-000/c057df1c945f28b5874144f25b748c8b . It's just a minor bug fix in the h264 decoding, many.gitignore files and the version will be renamed. The installation as a separate module should still work as described.

Reply:
I will stick to ffmpeg from the kodi tree for now. the others I can update